### PR TITLE
Document connection pool sizes and their overrides

### PR DESCRIPTION
**Pull request by mziolekda**

### CODE REVIEW

File: docs/2.8.0/docs/canton/usermanual/persistence.rst [hunk 1]
```diff
@@ -505,15 +505,49 @@ The number of parallel indexer connections can be configured via
...
 
 A Canton participant node will establish up to ``X + Y + 2`` permanent connections with the database, whereas a domain node
 will use up to ``X`` permanent connections, except for a sequencer with HA setup that will allocate up to ``2X`` connections. During
 startup, the node will use an additional set of at most ``X`` temporary connections during database initialisation.
 
-Please note that this number represents an upper bound of permanent connections and can be divided internally for different purposes,
+The number ``X`` represents an upper bound of permanent connections and is divided internally for different purposes,
 depending on the implementation. Consequently, the actual size of the write connection pool, for example, could be smaller.
 
+Following table summarizes the detailed split of the connection pools in different canton nodes. ``R`` signifies a *read* pool, ``W``
```
Comment (loiskilberg-da) [inline docs/2.8.0/docs/canton/usermanual/persistence.rst:517]:
```suggestion
The following table summarizes the detailed split of the connection pools in different Canton nodes. ``R`` signifies a *read* pool, ``W``
```

File: docs/2.8.0/docs/canton/usermanual/persistence.rst [hunk 2]
```diff
@@ -505,15 +505,49 @@ The number of parallel indexer connections can be configured via
...
+a *write* pool and ``A`` a *ledger api* pool, ``RW`` a combined *read/write* pool and ``EW`` an *exclusive writer* pool.
```
Comment (loiskilberg-da) [inline docs/2.8.0/docs/canton/usermanual/persistence.rst:518]:
```suggestion
a *write* pool , ``A`` a *ledger api* pool, ``RW`` a combined *read/write* pool, and ``EW`` an *exclusive writer* pool.
```

File: docs/2.8.0/docs/canton/usermanual/persistence.rst [hunk 3]
```diff
@@ -505,15 +505,49 @@ The number of parallel indexer connections can be configured via
...
++--------------+-------------------+--------------------+
+| Sequencer        | R = Z / 2          | column 3           |
+| exclusive writer | W = Z / 2          | column 3           |
++--------------+-------------------+--------------------+
+| Domain manager   | R = X / 2          |                    |
+|                  | W = X / 2 - 1      |                    |
+|                  | EW = 1             |                    |
++--------------+-------------------+--------------------+
+
+The results of the divisions are always rounded down, unless the result yields a zero. In that
```
Comment (loiskilberg-da) [inline docs/2.8.0/docs/canton/usermanual/persistence.rst:546]:
```suggestion
The results of the divisions are always rounded down unless the result yields a zero. In that
```

File: docs/2.8.0/docs/canton/usermanual/persistence.rst [hunk 4]
```diff
@@ -505,15 +505,49 @@ The number of parallel indexer connections can be configured via
...
+case a minimal pool size of 1 is ascertained.
```
Comment (loiskilberg-da) [inline docs/2.8.0/docs/canton/usermanual/persistence.rst:547]:
```suggestion
case, a minimal pool size of 1 is ascertained.
```

File: docs/2.8.0/docs/canton/usermanual/persistence.rst [hunk 5]
```diff
@@ -505,15 +505,49 @@ The number of parallel indexer connections can be configured via
...
+
+The sizes of the
```
Comment (loiskilberg-da) [inline docs/2.8.0/docs/canton/usermanual/persistence.rst:549]: wondering if there needs to be more to this sentence?

**Review by loiskilberg-da — CHANGES_REQUESTED** [review]
just a few suggestions

File: docs/2.8.0/docs/canton/usermanual/persistence.rst (hunk 5 above)
Comment (mziolekda) [reply docs/2.8.0/docs/canton/usermanual/persistence.rst:549]: It was a wip, more is coming

File: docs/2.8.0/docs/canton/usermanual/persistence.rst [hunk 6]
```diff
@@ -505,14 +505,84 @@ The number of parallel indexer connections can be configured via
...
+    canton.sequencers.<sequencer-name>.sequencer.high-availability.exclusive-storage.max-connections = Z
 
 A Canton participant node will establish up to ``X + Y + 2`` permanent connections with the database, whereas a domain node
 will use up to ``X`` permanent connections, except for a sequencer with HA setup that will allocate up to ``2X`` connections. During
 startup, the node will use an additional set of at most ``X`` temporary connections during database initialisation.
 
-Please note that this number represents an upper bound of permanent connections and can be divided internally for different purposes,
+The number ``X`` represents an upper bound of permanent connections and is divided internally for different purposes,
 depending on the implementation. Consequently, the actual size of the write connection pool, for example, could be smaller.
+Some of the allotted connections will be taken by the *read* pool, some by the *write* pool and a single additional connection
```
Comment (loiskilberg-da) [inline docs/2.8.0/docs/canton/usermanual/persistence.rst:522]:
```suggestion
Some of the allotted connections will be taken by the *read* pool, some will be taken by the *write* pool, and a single additional connection
```

File: docs/2.8.0/docs/canton/usermanual/persistence.rst [hunk 7]
```diff
@@ -505,14 +505,84 @@ The number of parallel indexer connections can be configured via
...
+::
+
+    canton.sequencers.sequencer.storage.parameters.connection-allocation...
+    canton.mediators.mediator.storage.parameters.connection-allocation...
+    canton.domain-managers.domain_manager.storage.parameters.connection-allocation...
+
+Where a node operates a combined *read/write* connection pool, the numbers for ``R`` and ``W`` overwrites are added
+together to determine the overall pool size.
+
+The effective connection pool sizes are reported by the Canton nodes at start-up
```
Comment (loiskilberg-da) [inline docs/2.8.0/docs/canton/usermanual/persistence.rst:581]:
```suggestion
The effective connection pool sizes are reported by the Canton nodes at startup.
```

File: docs/3.0.0/docs/canton/usermanual/persistence.rst [hunk 8]
```diff
@@ -505,14 +505,84 @@ The number of parallel indexer connections can be configured via
...
+    canton.sequencers.<sequencer-name>.sequencer.high-availability.exclusive-storage.max-connections = Z
 
 A Canton participant node will establish up to ``X + Y + 2`` permanent connections with the database, whereas a domain node
 will use up to ``X`` permanent connections, except for a sequencer with HA setup that will allocate up to ``2X`` connections. During
 startup, the node will use an additional set of at most ``X`` temporary connections during database initialisation.
 
-Please note that this number represents an upper bound of permanent connections and can be divided internally for different purposes,
+The number ``X`` represents an upper bound of permanent connections and is divided internally for different purposes,
 depending on the implementation. Consequently, the actual size of the write connection pool, for example, could be smaller.
+Some of the allotted connections will be taken by the *read* pool, some by the *write* pool and a single additional connection
```
Comment (loiskilberg-da) [inline docs/3.0.0/docs/canton/usermanual/persistence.rst:522]:
```suggestion
Some of the allotted connections will be taken by the *read* pool, some will be taken by the *write* pool, and a single additional connection
```

File: docs/3.0.0/docs/canton/usermanual/persistence.rst [hunk 9]
```diff
@@ -505,14 +505,84 @@ The number of parallel indexer connections can be configured via
...
+::
+
+    canton.sequencers.sequencer.storage.parameters.connection-allocation...
+    canton.mediators.mediator.storage.parameters.connection-allocation...
+    canton.domain-managers.domain_manager.storage.parameters.connection-allocation...
+
+Where a node operates a combined *read/write* connection pool, the numbers for ``R`` and ``W`` overwrites are added
+together to determine the overall pool size.
+
+The effective connection pool sizes are reported by the Canton nodes at start-up
```
Comment (loiskilberg-da) [inline docs/3.0.0/docs/canton/usermanual/persistence.rst:581]:
```suggestion
The effective connection pool sizes are reported by the Canton nodes at startup.
```

**Review by loiskilberg-da — COMMENTED** [review]
just a couple more suggestions :)

**Pull request merged**